Docs: Unify API of the `autocomplete` and `label` props

dlouhak opened the pull request.

literat commented on packages/web-react/src/components/TextFieldBase/README.md:
```diff
@@ -43,24 +43,25 @@ TextFieldBase with password toggle (button to reveal the password):
+| `isLabelHidden`     | `boolean`                                                     | –       | no       | Whether is label hidden                                                 |
+| `isMultiline`       | `boolean`                                                     | –       | no       | Whether is DOM element `textarea`                                       |
+| `isRequired`        | `boolean`                                                     | –       | no       | Whether is field required                                               |
+| `label`             | `string`                                                      | –       | yes      | Label text                                                              |
```
I am afraid that the label is not technically required by the types.

dlouhak replied:
[If I understand you well](https://github.com/lmc-eu/spirit-design-system/pull/957/commits/8540af8929b40e8c5c6bafda21ee1b4c6647ff5b#r1276716966), currently it's not possible define `label` as required prop in the React? Why?

literat replied:
It is technically possible, but it is a BC, IMHO.

dlouhak replied:
Ah, you're right. I reverted those changes, so we/you could fix this in another PR.

literat replied:
Thanks :-)

literat commented on packages/web-react/src/components/TextField/README.md:
```diff
@@ -46,23 +46,23 @@ TextField with password toggle (button to reveal the password):
+| `isDisabled`        | `boolean`                                                     | –       | no       | Whether is field disabled                                               |
+| `isLabelHidden`     | `boolean`                                                     | –       | no       | Whether is label hidden                                                 |
+| `isRequired`        | `boolean`                                                     | –       | no       | Whether is field required                                               |
+| `label`             | `string`                                                      | –       | yes      | Label text                                                              |
```
The label is not technically required.

crishpeen replied:
Technically not, but we want it to be filled every time. :)

literat replied:
Is this a fix or BC?

literat replied:
@adamkudrna This is React and if the `label` is marked as required it should be the same in the types.

literat reviewed (commented):
We should fix types for the label prop.

adamkudrna reviewed (commented):
Thank you for caring and all those tiny fixes and improvements 🙏🏻.

Just one thing: the `autocomplete` attribute is not part of our code which is (should) be the reason why it's not present in (most of) the docs. It is forwarded just like any other property to the native HTML element.

This leaves me with two options:

1. Don't mention `autocomplete` anywhere and keep relying on prop forwarding (transferring).
2. Re-introduce the discussion about documenting props that are not in code, but frequently used.

dlouhak commented:
> Thank you for caring and all those tiny fixes and improvements 🙏🏻.
> 
> Just one thing: the `autocomplete` attribute is not part of our code which is (should) be the reason why it's not present in (most of) the docs. It is forwarded just like any other property to the native HTML element.
> 
> This leaves me with two options:
> 
> 1. Don't mention `autocomplete` anywhere and keep relying on prop forwarding (transferring).
> 2. Re-introduce the discussion about documenting props that are not in code but still frequently used.

It's my pleasure. ✨ 

Agree with you that those HTML attributes are not directly part of the Spirit's components, so there shouldn't be a need to mention them in the docs, but all of the attributes (except `data-*`) have to be explicitly allowed in the Twig implementation (meanwhile React will pass them as rest props if I understand the code well).

This could be a little bit confusing for developers and without looking inside the component implementation, they couldn't know why they can pass eg. `autocomplete` but not `list` attribute.

I would be inclined to the second option so I've added this topic at Thursday's meeting. 🥌

dlouhak commented:
> > Thank you for caring and all those tiny fixes and improvements 🙏🏻.
> > Just one thing: the `autocomplete` attribute is not part of our code which is (should) be the reason why it's not present in (most of) the docs. It is forwarded just like any other property to the native HTML element.
> > This leaves me with two options:
> > 
> > 1. Don't mention `autocomplete` anywhere and keep relying on prop forwarding (transferring).
> > 2. Re-introduce the discussion about documenting props that are not in code but still frequently used.
> 
> It's my pleasure. ✨
> 
> Agree with you that those HTML attributes are not directly part of the Spirit's components, so there shouldn't be a need to mention them in the docs, but all of the attributes (except `data-*`) have to be explicitly allowed in the Twig implementation (meanwhile React will pass them as rest props if I understand the code well).
> 
> This could be a little bit confusing for developers and without looking inside the component implementation, they couldn't know why they can pass eg. `autocomplete` but not `list` attribute.
> 
> I would be inclined to the second option so I've added this topic at Thursday's meeting. 🥌

Following on from #DS-836 and #993, we'll mention all of the passing HTML attributes in the component documentation.

I reverted changes with required label in React. Will be solved during #DS-888.

literat reviewed (approved):
Well Done 👍
